core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -461,6 +462,7 @@ public synchronized boolean isReady() {
 
       @Override
       public void cancel(Status reason) {
+        reason = stripCause(reason);
```

Comment:
@ejona86 yep, but I think it's the other way around. `internalCancel` should be stripped of the cause. It's confusing 😅 .